### PR TITLE
Use managed identities for Postgres connection

### PR DESCRIPTION
**Pull request by paulsimonandrews**
https://eaflood.atlassian.net/browse/PSD-916

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,6 +27,10 @@ ENTRYPOINT dotnet watch --project ./FFCDemoPaymentService run --urls "http://*:$
 FROM defradigital/dotnetcore:${PARENT_VERSION} AS production
 ARG PARENT_VERSION
 LABEL uk.gov.defra.ffc.parent-image=defradigital/dotnetcore:${PARENT_VERSION}
+
+USER root
+RUN apk --no-cache add postgresql-client
```
Comment (johnwatson484) [inline Dockerfile:32]: What role does this package perform?  Is it something we need in every postgres container or is it specific to .NET Core?

File: FFCDemoPaymentService/Data/ApplicationDbContext.cs [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+        public ApplicationDbContext(ConnectionStringBuilder builder)
+            : base()
+        {
+            this.builder = builder;
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
+            string connectionString = Task.Run(builder.GetConnectionString).Result;
+            System.Console.WriteLine($"Using connection string ending: {connectionString.Substring(connectionString.Length - 10)}");
```
Comment (johnwatson484) [inline FFCDemoPaymentService/Data/ApplicationDbContext.cs:20]: We can probably remove this now unless you feel it is useful to help us monitor in the short term?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,3 +1,81 @@
 @Library('defra-library@v-8') _
 
-buildDotNetCore environment: 'dev', project: 'FFCDemoPaymentService'
+//buildDotNetCore environment: 'dev', project: 'FFCDemoPaymentService'
+
+def config= [environment: 'dev', project: 'FFCDemoPaymentService']
```
Comment (johnwatson484) [inline Jenkinsfile:5]: Needs to be restored before merging

File: helm/ffc-demo-payment-service-core/values.yaml [hunk 4]
```diff
@@ -36,7 +36,7 @@ container:
   requestCpu: 40m
   limitMemory: 120Mi
   limitCpu: 80m
-  readOnlyRootFilesystem: true
+  readOnlyRootFilesystem: false
```
Comment (johnwatson484) [inline helm/ffc-demo-payment-service-core/values.yaml:39]: Should be returned to read only file system

File: helm/ffc-demo-payment-service-core/values.yaml [hunk 5]
```diff
@@ -46,9 +46,11 @@ container:
   comPlusEnableDiagnostics: 0
 
 postgresService:
-  postgresConnectionString: User Id=postgres;Server=ffc-demo-payment-core-postgres;Port=5432;Database=ffc_demo_payments;Password=ppp;
+  postgresConnectionStringAlt: dummy
+  postgresConnectionString: dummy
```
Comment (johnwatson484) [inline helm/ffc-demo-payment-service-core/values.yaml:50]: Is there a need to split these or is this left over from testing?

File: Dockerfile (hunk 1 above)
Comment (paulsimonandrews) [reply Dockerfile]: This was just for testing to get `psql` installed on the pod. Now removed.

File: Jenkinsfile (hunk 3 above)
Comment (paulsimonandrews) [reply Jenkinsfile]: Done.

File: helm/ffc-demo-payment-service-core/values.yaml (hunk 5 above)
Comment (paulsimonandrews) [reply helm/ffc-demo-payment-service-core/values.yaml]: Connection string has been replaced by individual username, password, host etc settings for the DB

File: helm/ffc-demo-payment-service-core/values.yaml (hunk 4 above)
Comment (paulsimonandrews) [reply helm/ffc-demo-payment-service-core/values.yaml]: Done

File: FFCDemoPaymentService/Data/ApplicationDbContext.cs (hunk 2 above)
Comment (paulsimonandrews) [reply FFCDemoPaymentService/Data/ApplicationDbContext.cs]: Done. It logs every 10s, so too much to be left in!

**Pull request merged**